server/matchmaker/algorithm/team_matchmaker.py

```diff
@@ -0,0 +1,286 @@
+import statistics
+from collections import defaultdict, namedtuple
+from typing import Dict, Iterable, List, Tuple
+
+from sortedcontainers import SortedList
+
+from ...config import config
+from ...decorators import with_logger
+from ..search import CombinedSearch, Match, Search, get_average_rating
+from .matchmaker import Matchmaker
+
+Game = namedtuple('Game', ('match', 'quality'))
```

Comment:
I prefer `typing.NamedTuple`. There should be some examples in various `typedef.py` files.